src/meta/src/controller/fragment.rs

```diff
@@ -786,12 +860,51 @@ impl CatalogController {
 
         let mut table_fragments = BTreeMap::new();
         for job in jobs {
-            let fragment_actors = FragmentModel::find()
-                .find_with_related(Actor)
+            let fragments = FragmentModel::find()
                 .filter(fragment::Column::JobId.eq(job.job_id))
                 .all(&inner.db)
                 .await?;
 
+            let fragment_actors = {
+                let guard = self.env.shared_actor_infos().read_guard();
```

Comment:
Seems that in `get_job_fragments_by_id` there is similar logic to read from shared actor info to compose the ActorModel of a fragment. Can we extract the logic to a common method of shared actor info?